jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannelOverHttp.java

```diff
@@ -111,7 +118,14 @@ public boolean needContent()
         {
             if (LOG.isDebugEnabled())
                 LOG.debug("produceContent has no content, parsing and filling");
-            _httpConnection.parseAndFillForContent();
+            try
+            {
+                _httpConnection.parseAndFillForContent();
+            }
+            catch (Throwable x)
+            {
+                _content = new HttpInput.ErrorContent(x);
+            }
```

Comment:
Please refactor into a common method the two occurrences.

Reply:
Done.